avocado/utils/genio.py

```diff
@@ -174,3 +174,22 @@ def write_one_line(filename, line):
     :type line: str
     """
     write_file(filename, line.rstrip('\n') + '\n')
+
+
+def safe_write_file(filename, data):
+    """
+    Write data to a file.
```

Comment:
Sure, I haven't investigated which exceptions it raises. It looks like os.write raises OSError. Not sure whether it also raises IOError, if so you should catch both.

Anyway in general in some cases it's also fine to catch `Exception` and in very rare cases it's even fine to catch any kind of exception. We do this eg. in test: https://github.com/avocado-framework/avocado/blob/master/avocado/core/test.py#L471

Anyway always try to keep the number of exceptions minimal so if OSError suffices, use it. We can always expand it if needed.